core/src/avm1/globals/bitmap_data.rs

```
@@ -0,0 +1,1120 @@
+    }
+
+    Ok(this.as_bitmap_data_object().unwrap().get_height().into())
+}
+
+pub fn get_width<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let bitmap_data = this.as_bitmap_data_object().unwrap();
+
+    if bitmap_data.get_disposed() {
+        return Ok((-1).into());
+    }
+
+    Ok(this.as_bitmap_data_object().unwrap().get_width().into())
+}
+
+pub fn get_transparent<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let bitmap_data = this.as_bitmap_data_object().unwrap();
+
+    if bitmap_data.get_disposed() {
+        return Ok((-1).into());
+    }
+
+    Ok(this
+        .as_bitmap_data_object()
+        .unwrap()
+        .get_transparency()
+        .into())
+}
+
+pub fn get_rectangle<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let bitmap_data = this.as_bitmap_data_object().unwrap();
+
+    if bitmap_data.get_disposed() {
+        return Ok((-1).into());
+    }
+
+    let proto = activation.context.system_prototypes.rectangle_constructor;
+    let rect = proto.construct(
+        activation,
+        &[
+            0.into(),
+            0.into(),
+            bitmap_data.get_width().into(),
+            bitmap_data.get_height().into(),
+        ],
+    )?;
+    Ok(rect.into())
+}
+
+pub fn get_pixel<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    let bitmap_data = this.as_bitmap_data_object().unwrap();
+
+    if bitmap_data.get_disposed() {
+        return Ok((-1).into());
+    }
+
+    let x = args.get(0).and_then(|x| x.coerce_to_i32(activation).ok());
+    let y = args.get(1).and_then(|x| x.coerce_to_i32(activation).ok());
```

`ok()` swallows an error, instead there should be `?` somewhere, here's a pathological example:
```
var o = {valueOf: function() {
	throw "BAD";
	return 3;
}};
trace(bData.getPixel(o, o));
```